services/bulk_fix.py

```diff
@@ -0,0 +1,130 @@
+import pymongo
+import logging
+import json
+from bson.objectid import ObjectId
+import base64
+from api.models import DJ_Instrument, Instrument
+from api_auth.models import InstrumentAuth
+from django.contrib.auth.models import User
+
+
+SUBS = {'BMI:BMI': 'BMI:BMI',
+ 'BMI:Height': 'BMI:HEIGHT',
+ 'BMI:RecentHeight': 'BMI:RECENT_HEIGHT',
+ 'BMI:RecentLowBMI': 'BMI:RECENT_LOW_BMI',
+ 'BMI:RecentWeight': 'BMI:RECENT_WEIGHT',
+ 'BMI:Weight': 'BMI:WEIGHT',
+ 'BingeEating:OBEs per month': 'Binge Eating:OBES_PER_MONTH',
+ 'BingeEating:OBEs per week': 'Binge Eating:OBES_PER_WEEK',
+ 'BingeEating:SBEs per month': 'Binge Eating:SBES_PER_MONTH',
+ 'BingeEating:SBEs per week': 'Binge Eating:SBES_PER_WEEK',
+ 'Diuretics:Average_number_per_month': 'DIURETICS:AVERAGE_NO_MONTH',
+ 'Diuretics:Average_number_per_week': 'DIURETICS:AVERAGE_NO_WEEK',
+ 'Exercise:Average_number_episodes_per_month': 'EXERCISE:AVERAGE_PER_MONTH',
+ 'Exercise:Average_number_episodes_per_week': 'EXERCISE:AVERAGE_PER_WEEK',
+ 'Exercise:Average_number_mins_per_episode': 'EXERCISE:AVERAGE_MINUTES_PER',
+ 'Exercise:Type': 'EXERCISE:TYPE',
+ 'Interview:Date:date': 'INTERVIEW:DATE:date',
+ 'Interview:InterviewerID': 'INTERVIEW:INTERVIEW_ID',
+ 'Interview:SubjectAge:number': 'INTERVIEW:SUBJECT_AGE:number',
+ 'Interview:SubjectID': 'INTERVIEW:SUBJECT_ID',
+ 'Laxatives:Average_number_per_month': 'LAXATIVES:AVERAGE_NO_MONTH',
+ 'Laxatives:Average_number_per_week': 'LAXATIVES:AVERAGE_NO_WEEK',
+ 'OtherMethod:Average_number_per_month': 'OTHER_METHOD:AVERAGE_NO_MONTH',
+ 'OtherMethod:Average_number_per_week': 'OTHER_METHOD:AVERAGE_NO_WEEK',
+ 'OtherMethod:Name': 'OTHER_METHOD:NAME',
+ 'Vomiting:Average_number_per_month': 'VOMITING:AVERAGE_NO_MONTH',
+ 'Vomiting:Average_number_per_week': 'VOMITING:AVERAGE_NO_WEEK'}
+
+
+def find_all_questionnaires():
+###
+# find all distinct questionnaires by major_version
+# and return their major version and shortname
+###
+    client = pymongo.MongoClient('mongodb')
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code might contain a resource leak. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the following resource: *client*. The resource is allocated by call *pymongo.MongoClient*. Execution paths that do not contain closure statements were detected. Close *client* in a try-finally block to prevent the resource leak.